### PR TITLE
Adding some logic to handle different platforms and compilers

### PR DESCRIPTION
**Pull request by dfm**
This adds some generally untested logic based on @lgarrison's suggested compiler flags that tries to set them sensibly based on the environment.

Here are the results I get when running the benchmark on my M1 apple laptop:

```
Running with N=3554, M=100000, dtype float64
baseline took 5.052 sec
astropy took 11.71 sec
frac isclose 100%
max frac err 3.865e-09
```

### CODE REVIEW

**Comment by lgarrison** [comment]
Thanks, this is great! I knew there must be a right way to organize this, but I wasn't sure what it was.

I think in the end most/all of the commented-out flags ended up not helping (or even hurting, in the case of `-funroll-loops`), but I will double check this.  Need to un-break my environment first...

Also I see you noticed I mashed together the cflags and ldflags like a Bad Programmer 😅

File: setup.py [hunk 1]
```diff
@@ -1,45 +1,72 @@
...
+
+            # None of the rest of these will work on Mac
+            if not MACOS:
+                cc = " ".join(self.compiler.compiler_so[:2])
+                
+                # TODO: this can give a huge speedup when using the Intel compiler on an AMD CPU
+                # but if we're on an Intel CPU, we want -march=native instead. How to detect this?
+                #if "icc" in cc:
+                #    if has_flag(self.compiler, "-march=core-avx2"):
+                #        cflags += ["-march=core-avx2"]
```
Comment (lgarrison) [inline setup.py:39]: I think we'll just stick to Intel CPUs for now, can solve this later

File: setup.py [hunk 2]
```diff
@@ -1,45 +1,72 @@
...
+                    # These are optimistic/untested commands that might work if one has
+                    # an SVML lib available for Clang to link against
+                    #cflags += ["-fveclib=SVML"]
+                    #ldflags += ["-lsvml"]
+
+                # TODO: AOCC on Clang 13 seems to only support C++11
+                # but the Pybind11 auto-detection doesn't seem to notice
+                #if "aocc" in cc:  # TODO: the AOCC executable is actually just named `clang`...
+                #    cflags += ["-std=c++11"]
+                #    ldflags += ["-lamdlibm", "-lm"]  # not sure if these help...
```
Comment (lgarrison) [inline setup.py:53]: AOCC didn't work any magic performance-wise, so I think it's okay to leave this broken.

**Comment by dfm** [comment]
LGTM!

**Pull request merged**